x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.ts

```diff
@@ -106,25 +139,82 @@ export const isolationRequestHandler = function (
     caseIDs = [...new Set(caseIDs)];
 
     // create an Action ID and dispatch it to ES & Fleet Server
-    const esClient = context.core.elasticsearch.client.asCurrentUser;
+    let esClient = context.core.elasticsearch.client.asInternalUser;
     const actionID = uuid.v4();
-    let result;
+
+    let fleetActionIndexResult;
+    let logsEndpointActionsResult;
+
+    const agents = endpointData.map((endpoint: HostMetadata) => endpoint.elastic.agent.id);
+    const doc = {
+      '@timestamp': moment().toISOString(),
+      agent: {
+        id: agents,
+      },
+      EndpointAction: {
+        action_id: actionID,
+        expiration: moment().add(2, 'weeks').toISOString(),
+        type: 'INPUT_ACTION',
+        input_type: 'endpoint',
+        data: {
+          command: isolate ? 'isolate' : 'unisolate',
+          comment: req.body.comment ?? undefined,
+        },
+      } as Omit<EndpointAction, 'agents' | 'user_id'>,
+      user: {
+        id: user!.username,
+      },
+    };
+
+    // if .logs-endpoint.actions data stream exists
+    // create action request record in .logs-endpoint.actions DS as the user
+    const doesLogsEndpointActionsDsExist = await doLogsEndpointActionDsExists({
+      esClient,
+      logger: endpointContext.logFactory.get('host-isolation'),
+      dataStreamName: ENDPOINT_ACTIONS_DS,
+    });
+    if (doesLogsEndpointActionsDsExist) {
+      esClient = context.core.elasticsearch.client.asCurrentUser;
+      try {
+        logsEndpointActionsResult = await esClient.index<LogsEndpointAction>({
+          index: `${ENDPOINT_ACTIONS_DS}-default`,
+          body: {
+            ...doc,
+          },
+        });
+      } catch (e) {
+        return res.customError({
+          statusCode: 500,
+          body: { message: e },
+        });
+      }
+
+      if (logsEndpointActionsResult.statusCode !== 201) {
+        return res.customError({
+          statusCode: 500,
+          body: {
+            message: logsEndpointActionsResult.body.result,
+          },
+        });
+      }
+    }
+
+    // create action request record as system user in .fleet-actions
     try {
-      result = await esClient.index<EndpointAction>({
+      // we use this check to ensure the user has permission to write to this new index
+      // and thus allow this action to be added to the fleet index by kibana
+      if (!doesLogsEndpointActionsDsExist) {
+        esClient = context.core.elasticsearch.client.asCurrentUser;
```

Comment:
i.e. this can be a constant.

Reply:
e1e3d4e2fc123b1a2dcba7f84dbe7ad23c4bdbff